bsoid_app.py

```diff
@@ -61,21 +57,19 @@ def get_env_variable(var_name, default_value=None):
 [features, _] = load_feats(WORKING_DIR, PREFIX)
 [sampled_features, _] = load_embeddings(WORKING_DIR, PREFIX)
 [_, assignments, _, _] = load_clusters(WORKING_DIR, PREFIX)
-learning_protocol = machine_learner.protocol(WORKING_DIR, PREFIX, features, sampled_features, assignments)
+learning_protocol = machine_learner.Protocol(WORKING_DIR, PREFIX, features, sampled_features, assignments)
 learning_protocol.main()
 
 [ROOT_PATH, DATA_DIRECTORIES, FRAMERATE, pose_chosen, input_filenames, _, processed_input_data, _] \
     = load_data(WORKING_DIR, PREFIX)
 [_, _, _, clf, _, _] = load_classifier(WORKING_DIR, PREFIX)
-creator = video_creator.creator(ROOT_PATH, DATA_DIRECTORIES, processed_input_data, pose_chosen,
-                                WORKING_DIR, PREFIX, FRAMERATE, clf, input_filenames)
-creator.main()
-
-[ROOT_PATH, DATA_DIRECTORIES, FRAMERATE, pose_chosen, input_filenames, _, processed_input_data, _] \
-    = load_data(WORKING_DIR, PREFIX)
-[_, _, _, clf, _, predictions] = load_classifier(WORKING_DIR, PREFIX)
-predictor = predict.prediction(ROOT_PATH, DATA_DIRECTORIES, input_filenames, processed_input_data, WORKING_DIR,
-                               PREFIX, FRAMERATE, pose_chosen, predictions, clf)
-predictor.main()
 
-streamlit_run('./bsoid_app/bsoid_analysis')
+for directory in DATA_DIRECTORIES:
+    full_directory_path = os.path.join(ROOT_PATH, directory)
+    # Create a new instance of Creator for the current directory
+    creator = video_creator.Creator(
```

Comment:
Why do we have to pass the `full_directory_path` twice?

Reply:
the first instance should actually pass DATA_DIRECTORIES. edited.